### PR TITLE
Bitmap alpha format support

### PR DESCRIPTION
**Pull request by MarchingCube**
## What does the pull request do?

Skia and D2D backends hardcode alpha formats for bitmaps loaded directly from memory. This PR is trying to fix that by allowing users to provide alpha format.

## How was the solution implemented (if it's not obvious)?

Direct2D and Skia backends now require alpha format to be provided when loading bitmaps from memory. Writeable bitmap also accepts alpha formats and will default to premultiplied alpha.

## Checklist

- [ ] Added unit tests (if possible)?
- [ ] Added XML documentation to any related classes?
- [ ] Consider submitting a PR to https://github.com/AvaloniaUI/Avaloniaui.net with user documentation

### CODE REVIEW

**Comment by kekekeks** [comment]
It seems that GPU-accelerated SKCanvas ignores the alpha-type on bitmaps.

**Comment by MarchingCube** [comment]
Bug in Skia it seems, after updating to latest SkiaSharp EAP (2.84.0-pr.1321.3 currently) it works correctly.

**Comment by jmacato** [comment]
@MarchingCube any updates? :)

**Comment by MarchingCube** [comment]
> @MarchingCube any updates? :)

I guess until we update Skia we can't merge this in. Or we would need to implement some kind of fallback but I don't like that idea since this is fixed properly in newer Skia.

**Comment by MarchingCube** [comment]
Waiting for https://github.com/AvaloniaUI/Avalonia/pull/4222.

**Comment by MarchingCube** [comment]
@kekekeks After updating to new Skia everything works fine. Had to change theme in render demo since I couldn't see sidebar anymore.

File: samples/RenderDemo/App.xaml
Comment (kekekeks) [inline samples/RenderDemo/App.xaml:2]: @danwalmsley I think default theme got just broken somehow.

**Comment by adirh3** [comment]
@MarchingCube Any way this will be part of 0.10 preview 2? :)

**Comment by MarchingCube** [comment]
@adirh3 I've updated this PR to latest master and fixed conflicts. Can you actually test it to see if it fixes your problem?

**Comment by adirh3** [comment]
> @adirh3 I've updated this PR to latest master and fixed conflicts. Can you actually test it to see if it fixes your problem?

Thanks, just checked and it fixed my issue (#4103) when changed AlphaFormat to Unpremul.

**Comment by MarchingCube** [comment]
@danwalmsley @kekekeks Updated this PR to remove API break in non-platform interface part. There are still changes in the rendering interface, but this part is pretty much internal so I hope it won't cause problems.

**Review by kekekeks — APPROVED** [review]

**Pull request merged**
